### ISSUE TITLE
Bug in _get_price_data_binance

### ISSUE THREAD

**Issue by scientes** [provinzio/CoinTaxman#48]
```
UTC_Time,Account,Operation,Coin,Change,Remark
2021-05-08 21:30:41,Spot,Buy,BURGER,3.00000000,""
2021-05-08 23:01:34,Spot,Transaction Related,BURGER,-3.00000000,""
```
leads to an infinte recursion:

```
2021-05-09 21:11:32,579 taxman       DEBUG    Starting evaluation...
2021-05-09 21:11:32,590 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BURGEREUR&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:33,140 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BURGERBTC&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:33,554 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:33,950 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:34,366 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:34,771 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:35,185 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:35,701 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:36,217 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:36,618 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:37,025 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:37,436 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:37,949 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:38,365 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
2021-05-09 21:11:39,295 price_data   DEBUG    Calling https://api.binance.com/api/v3/aggTrades?symbol=BTCBURGER&startTime=1620514864000&endTime=1620514924000
```

My idea why this is happening:
This fails due to only being tradeable with bnb and not btc thus breaking the normal get_price function

**Comment by sibbl** [comment]
I also experienced some strange behaviors in the Binance price API and how it's used in this project.

1. I think this project should raise exceptions instead of continuing with a default price of 0. This leads to a wrong exports and you only get an error displayed once because `0` will be stored in the database.
2. We currently only get the price in a  time window of 1 minute. Other algorithms like Kraken and now also Coinbase Pro try it with 5 minutes, then 10 minutes, 15 minutes etc... We should adapt the algorithm here as well.
3. However, there still might be cases where we don't get any data point within a one hour window - that maximum which the Binance API allows ([sample](https://api.binance.com/api/v3/aggTrades?symbol=BTTEUR&startTime=1617716395000&endTime=1617719995000)). In that case, we should trigger the same behavior like we do when we get "Invalid symbol.", shouldn't we?

Finally, if we did receive "Invalid symbol" or didn't get any data within the 1 hour interval, we might want to try other intermediate asset than BTC (which is the current default). BNB often works fine and if both BTC+BNB failed, I always were successful with BUSD.

However, I'm not familiar with the progress of #16 and would guess that this might also solve a lot of problems mentioned above.
